tests/components/myuplink/conftest.py

```diff
@@ -38,7 +38,29 @@ def mock_config_entry(expires_at: int) -> MockConfigEntry:
             "auth_implementation": DOMAIN,
             "token": {
                 "access_token": "Fake_token",
-                "scope": "READSYSTEM offline",
+                "scope": "WRITESYSTEM READSYSTEM offline_access",
+                "expires_in": 86399,
+                "refresh_token": "3012bc9f-7a65-4240-b817-9154ffdcc30f",
+                "token_type": "Bearer",
+                "expires_at": expires_at,
+            },
+        },
+        entry_id="myuplink_test",
+    )
+
+
+@pytest.fixture
+def mock_config_entry2(expires_at: int) -> MockConfigEntry:
```

Comment:
Modify the first mock config entry in the test instead with `hass.config_entries.async_update_entry`, before starting the reauth flow.